File: R/process/run_single_method.R
library(decoupleR)
suppressPackageStartupMessages(library(tidyverse))
source(file.path('R', 'process', 'methods_params.R'))

options( warn = -1 )

# Parse args
args <- commandArgs(trailingOnly = TRUE)
mat_path <- args[[1]]
net_path <- args[[2]]
method <- args[[3]]

if (method == 'gsea'){
  method <- 'fgsea'
}

# Methods dictionary
methods <- list(
  aucell = run_aucell,
  udt = run_udt,
  mdt = run_mdt,
  wmean = run_wmean,
  ulm = run_ulm,
  mlm = run_mlm,
  wsum = run_wsum,
  viper = run_viper,
  gsva = run_gsva,
  ora = run_ora,
  fgsea = run_fgsea
)

# Open
mat <- as.matrix(read.csv(mat_path))
colnames(mat) <- as.character(1:ncol(mat))
rownames(mat) <- as.character(1:nrow(mat))
net <- as_tibble(read.csv(net_path)) %>%
  mutate(source=as.character(source), target=as.character(target))

# Run
f <- methods[[method]]
params <- opts_list[[1]][[method]]
params[['mat']] <- mat
params[['net']] <- net
res <- do.call(f, params)

File: R/process/run_single_method.py
import decoupler as dc
import pandas as pd
import numpy as np
from methods_params import opts_list
import sys


# Parse args
args = sys.argv[1:]
mat_path = args[0]
net_path = args[1]
method = args[2]

# Methods dictionary
methods = {
  'aucell' : dc.run_aucell,
  'udt' : dc.run_udt,
  'mdt' : dc.run_mdt,
  'wmean' : dc.run_wmean,
  'ulm' : dc.run_ulm,
  'mlm' : dc.run_mlm,
  'wsum' : dc.run_wsum,
  'viper' : dc.run_viper,
  'gsva' : dc.run_gsva,
  'ora' : dc.run_ora,
  'gsea' : dc.run_gsea
}

# Open
mat = pd.read_csv(mat_path).T
mat.columns = (np.arange(mat.shape[1]) + 1).astype(str)
mat.index = (np.arange(mat.shape[0]) + 1).astype(str)
net = pd.read_csv(net_path)
net['source'] = net['source'].astype(str)
net['target'] = net['target'].astype(str)
net.columns = ['source','target','weight']

# Run
f = methods[method]
params = opts_list[method]
res = f(mat, net, **params)
